src/main/java/com/artipie/rpm/meta/XmlRepomd.java

```diff
@@ -72,15 +72,19 @@ public XmlRepomd(final Path path, final XmlFile xml) {
     /**
      * Begin repomd.
      * @param timestamp Current timestamp in seconds unix time.
-     * @throws XMLStreamException On error
+     * @throws IOException On error
      */
-    public void begin(final long timestamp) throws XMLStreamException {
-        this.xml.writeStartDocument(StandardCharsets.UTF_8.name(), "1.0");
-        this.xml.writeStartElement("repomd");
-        this.xml.writeDefaultNamespace("http://linux.duke.edu/metadata/repo");
-        this.xml.writeStartElement("revision");
-        this.xml.writeCharacters(String.valueOf(timestamp));
-        this.xml.writeEndElement();
+    public void begin(final long timestamp) throws IOException {
+        try {
+            this.xml.writeStartDocument(StandardCharsets.UTF_8.name(), "1.0");
+            this.xml.writeStartElement("repomd");
+            this.xml.writeDefaultNamespace("http://linux.duke.edu/metadata/repo");
+            this.xml.writeStartElement("revision");
+            this.xml.writeCharacters(String.valueOf(timestamp));
+            this.xml.writeEndElement();
+        } catch (final XMLStreamException ex) {
+            throw new IOException("Failed to start repomd", ex);
+        }
```

Comment:
@olenagerasimova it does not seem right to wrap `XMLStreamException` in `IOException`.  `XMLStreamException` is about logical errors and `IOException` is about I/O errors and no I/O happens in this method. It would be better to wrap `XMLStreamException`in some sort of `RuntimeException` pr just `Exception` if it has to be checked.

Reply:
@olegmoz we have a lot of places with such behavior and I've intoduced https://github.com/artipie/rpm-adapter/issues/323 to fix it